### PR TITLE
De-lint shell scripts with spellcheck and debug FORD upgrade

### PR DESCRIPTION
**Pull request by zbeekman**
FORD has introduced some sort of regression that causes it to fail when processing the unicode enabled JSON-Fortran... and maybe the normal JSON-Fortran...

There's no downside of always passing `--debug` to FORD, AFAICT, so I've added it to the CMake and build.sh scripts...

We should probably wait until FORD gets fixed to merge this... and also make sure that we re-trigger the CI builds after FORD is fixed, otherwise the documentation may be incorrect...

See https://github.com/cmacmackin/ford/issues/139#issuecomment-223824644 for more info....

### CODE REVIEW

**Comment by zbeekman** [comment]
BTW, this fixes #181

**Comment by jacobwilliams** [comment]
I'll wait to merge until you give the go ahead. Although it looks like all checks have failed. :)

**Comment by zbeekman** [comment]
Actually, the only thing failing is [building the documentation with FORD](https://travis-ci.org/jacobwilliams/json-fortran/jobs/135456701#L4501-L4528) (>= 4.6.0) due to a bug that I have reported upstream to @cmacmackin cmacmackin/ford#139

The reason that it's failing as well is that the documentation build step is within the build.sh script. As far as I see it there are three options to move this PR forward:

1. Wait for an upstream fix to merge this PR
2. Downgrade FORD 4.5.4 on Travis-CI
3. Build documentation for non-unicode enabled FORD (this will effectively build a subset of the documentation since enabling unicode only adds additional code and documentation) since FORD 4.6.1 seems to do alright if you don't enable unicode support.
4. Split out the documentation building into the `after_success` block. This way all the tests will run and pass, but if the documentation fails to build it will either error, or just continue on without building and deploying the documentation.

I'm against 3, and leaning towards solution 2. I'd consider 4, but the downside of 4 is that the documentation won't deploy until it is fixed upstream or we combine it with 2, but it is still possible in the future that the documentation will fail to build, but since it doesn't error it may go un-noticed.

So, I'm thinking I add 2 to this PR? (I should also review when new functionality was added and update the project file to use some new bells and whistles...)

Thoughts @jacobwilliams ?

**Comment by zbeekman** [comment]
So I'll probably just specify FORD==4.5.4 for now... Unless you object

**Comment by zbeekman** [comment]
scratch that... see #205 I'll implement this now... this will resolve *ALL* the problems... some Homebrew issues blocking upgrading FORD, and make the JSON-Fortran documentation better.

BTW, for now, if you relaunch the checks for the documentation build on Travis-CI it should pass due to Chris' latest bug fixes...

**Comment by zbeekman** [comment]
@jacobwilliams Assuming all the tests pass, and you think everything looks good (note: I have not tested the Intel FPP yet... I think my license expired...) I believe we can :ship: this PR.

If you wouldn't mind creating a patch release after merging this I would appreciate it, so that I can get FORD updated in Homebrew, which will require updating JSON-Fortran in homebrew so that it doesn't choke when being compiled from source in the Homebrew build environment...

File: build.sh [hunk 1]
```diff
@@ -343,8 +346,9 @@ echo ""
 if [[ $JF_SKIP_DOCS != [yY]* ]]; then
     if hash ford 2>/dev/null; then
     echo "Building documentation..."
-    [[ $TRY_UNICODE = [yY]* ]] && MACRO_FLAG="-m USE_UCS4"
-    ford $MACRO_FLAG -p $PAGESDIR $FORDMD
+    [[ $TRY_UNICODE = [yY]* ]] && MACRO_FLAG=("-m" "USE_UCS4")
+    echo "$FPP" > preprocessor-def.md # Override via include in project file, until FORD gets CLI for this
+    ford --debug "${MACRO_FLAG[@]}" -p "$PAGESDIR" "$FORDMD"
```
Comment (zbeekman) [inline build.sh]: We may want to remove the `--debug` flag here... it generates a bunch of output... however, I want to leave it in the CMake build script to ease Homebrew debugging...

**Comment by jacobwilliams** [comment]
Will do shortly...

**Comment by jacobwilliams** [comment]
Verified that it works with ifort.

**Comment by jacobwilliams** [comment]
Should be done now. (note: I forgot to commit the updated release index for 5.0.1...so I just made a 5.0.2 release to correct it). Oops.

**Comment by zbeekman** [comment]
Awesome!

**Comment by jacobwilliams** [comment]
Something's wrong... it doesn't look like the updated documentation got deployed ([still says 5.0.0 here](http://jacobwilliams.github.io/json-fortran/index.html)).

**Comment by zbeekman** [comment]
Hmmm looks like problems even in 5.0.0 release... investigating...

**Pull request merged**